docs/proposals/parameter-distribution.md

```diff
@@ -0,0 +1,128 @@
+# Proposal for Parameter Distribution
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Design Details](#design-details)
+  - [Experiment API changes](#experiment-api-changes)
+  - [Correspondence for Katib Distributions and Framework Distributions](#correspondence-for-katib-distributions-and-framework-distributions)
+    - [Chocolate](#chocolate)
+    - [Goptuna](#goptuna)
+    - [Hyperopt](#hyperopt)
+    - [Optuna](#optuna)
+    - [Scikit-Optimize](#scikit-optimize)
+<!-- /toc -->
+
+## Summary
+This enhancement introduces `Distribution` to tuning parameters and remove redundantly `ParameterType`. 
+
+API field in the Experiment spec determine parameter type with distribution.  
+
+## Motivation
+Currently, Katib does not support determining a distribution for search space that samplers pick up parameters by users.
+
+Katib should be able to determine it by users since
+almost hyperparameter tuning algorithms (framework) can determine it by users.
+
+## Proposal
+We introduce a mechanism to determine a distribution for search space by users.
+That also means we introduce a mechanism to propagate distributions to suggestion-services and
+set them samplers of each framework.     
+
+## Design Details
+The proposal consists of a new Experiment API field and
+correspondence between Katib Distributions and Framework Distributions.
+
+### Experiment API changes
+We extend the Experiment API to introduce the new fields `Distribution` to configure the distribution for search space and
+remove the redundant fields `ParameterType`.
+
+```diff
+type ParameterSpec struct {
+	Name          string        `json:"name,omitempty"`
+- 	ParameterType ParameterType `json:"parameterType,omitempty"`
++ 	Distribution  Distribution  `json:"distribution,omitempty"`
```

Comment:
Great proposal @tenzen-y .  One question, how do we provide backward compatibility?

Reply:
@johnugeorge This is a good point.

For the time being (1~2 releases?), I think we can operate `ParameterType` and `Distribution` concurrently.
This means in the case of users determining `ParameterType`, suggestion-services operate as now; in the case of users determining `Distribution`, suggestion-services set distributions to sampler.

Also, we should add webhook validation to restrict `ParameterType` and `Distribution` so that only one of them is available. (`ParameterType` and `Distribution` are exclusive)

@andreyvelich @johnugeorge  wdyt?
If you agree with this, I will add this to the Proposal.

Reply:
SGTM. Also, add deprecation tag to `ParameterType`

Reply:
> Also, add deprecation tag to ParameterType

SGTM
I will add the tag to the following:

https://github.com/kubeflow/katib/blob/f941ec61e5fe8aa17919c51a8b347914ae70b876/pkg/apis/manager/v1beta1/api.proto#L83